libwasmvm/src/test_utils.rs

```diff
@@ -0,0 +1,89 @@
+#![cfg(test)]
+
+use cosmwasm_std::{Decimal, Uint128};
+use std::str::FromStr as _;
+
+/// Asserts that two expressions are approximately equal to each other.
+///
+/// The ratio argument defines how wide a range of values we accept, and is applied
+/// to the **second** (`right`) argument.
```

Comment:
One thing I hate about Rust testing is that there is no clear way where to put `exepected` and where to put `actual`. We usually put `actual` on the left in the cosmwasm codebase but even that is not completely consistent.

But in order to stay consistent with existing assert macros, I would avoid making a difference in the argument order.

[Here](https://en.wikipedia.org/wiki/Relative_change_and_difference) are some insights in relative difference
<img width="1052" alt="Bildschirmfoto 2022-09-07 um 13 27 29" src="https://user-images.githubusercontent.com/2603011/188868085-7c9641cf-0a62-4114-bdd7-cf298daab1f7.png">

The approx library uses `max(|x|, |y|)` to calculate the relative difference from it: https://github.com/brendanzab/approx/blob/v0.5.1/src/relative_eq.rs#L70-L77 I think we can do that too.

Reply:
Oh, interesting. I can do that. My one gripe is if an assertion fails, I'll no longer be able to provide an "expected range" of values, which feels like it could be useful for debugging.

Reply:
Then again, it's no big deal and you make fair points. Yeah, I'll do it.

Reply:
The advantage I see is you don't need to make use of the mixed type operator Uint128*Decimal that we need to get rid of sooner or later.

I think an error message a la _The value 100 and 119 differ by 15.9% but only 10% difference is allowed_ is still fine.

Reply:
> The advantage I see is you don't need to make use of the mixed type operator Uint128*Decimal that we need to get rid of sooner or later.

Don't we still need to do this?

```rust
                // Use a relative difference comparison
                abs_diff <= largest * max_relative
```

Where largest is still `Uint128` and `max_relative` is `Decimal`.

Reply:
Yaou can do `let relative = Decimal::from_ratio(a.abs_diff(b), max(a, b))` and then check `relative <= max_relative `